src/platform/Linux/KeyValueStoreManagerImpl.h

```diff
@@ -25,6 +25,14 @@
 
 #include <platform/Linux/CHIPLinuxStorage.h>
 
+#ifndef KVSDIR
```

Comment:
I suggest to put it in `src/platform/Linux/ChipPlatformConfig.h` and prefix the defines with `CHIP_CONFIG_`. And I would actually allow to override the full path, not just the dir.